docs/architecture/databaseUtil.md

```diff
@@ -0,0 +1,87 @@
+# DatabaseUtil
+
+The [DatabaseUtil](../../src/utils/databaseUtil.ts) abstracts
+writing and reading to the file system,
+as well as parsing and serializing the game data.
+
+For now it is assumed that there is only one save file.
+
+## Delete
+
+```mermaid
+sequenceDiagram
+  autonumber
+  participant A as Actor
+  participant D as DatabaseUtil
+  participant F as File System
+
+  A ->> D: delete()
+  D ->> F: unlink()
+  D ->> D: error: unknown | null
+  D ->> D: hasGameFile() <br> isOnDisk: boolean <br>
+  D ->> A: isOnDisk: boolean <br> error: unknown | null
+
+  alt encountered error
+    D ->> D: hasGameFile() <br> isFileOnDisk: boolean
+    D ->> A: isFileOnDisk: boolean <br> error: unknown
+  else
+    D ->> A: isFileOnDisk: true <br> error: null
+  end
+```
+
+## HasGameFile
+
+```mermaid
+sequenceDiagram
+  autonumber
+  participant A as Actor
+  participant D as DatabaseUtil
+  participant F as File System
+
+  A ->> D: hasGameFile()
+  D ->> F: exists()
+  F ->> D: isFileOnDisk: boolean
+  D ->> A: isFileOnDisk: boolean
```

Comment:
`hasGameFile` just returns a `boolean`. Should `isFileOnDisk` be here? Should `hasGameFile` have a `type` with `isFileOnDisk`?

Reply:
I thought of a scenario where this could get confusing in code, but I found an eslint rule to fix it and made [this ticket](https://github.com/normal-devs/klicker-knight/projects/1#card-77308986) to add it.

I also just like that eslint rule, and I'm glad it exists.